flake.nix

```diff
@@ -18,6 +18,11 @@
   } //
   (flake-utils.lib.eachDefaultSystem (system:
     let
+      # TODO(aseipp): Use dirtyRev and dirtyShortRev to record dirty checkout
+      # when we update `build.rs` to understand dirty checkouts.
+      gitRev = self.rev or "";
+      gitDate = builtins.substring 0 8 (self.lastModifiedDate or self.lastModified or "19700101");
```

Comment:
I'd prefer the date to be "" on failure, but I'm not sure whether `builtins.substring` will panic in that case. Since `gitRev` will be "", and `build.rs` recognizes that as a failure, it doesn't really matter.

Reply:
`builtins.substring` is basically just a way of taking a slice of another string; in the case of `""` it would just return `""` no matter the (positive) input ranges you provide.

Reply:
Thank you!